### PR TITLE
python3Packages.mkdocs-git-revision-date-localized-plugin: init at 1.2.0

### PR DESCRIPTION
**Pull request by totoroot**
###### Description of changes
Adds [mkdocs-git-revision-date-localized](https://github.com/timvink/mkdocs-git-revision-date-localized)

A [MkDocs](https://www.mkdocs.org/) plugin that enables displaying the date of the last git modification of a page

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Janik-Haag — APPROVED** [review]
LGTM; but can you do a rebase?

File: pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+, fetchFromGitHub
+, pythonOlder
+, gitpython
+, mkdocs
+, pytz
+}:
+
+buildPythonPackage rec {
+  pname = "mkdocs-git-revision-date-localized-plugin";
+  version = "1.2.0";
```
Comment (natsukium) [inline pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix:12]:
```suggestion
  version = "1.2.0";
  format = "setuptools";
```
Would you please add format explicitly?

File: pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    rev = "v${version}" ;
+    hash = "sha256-l7NUmsdd1FHkbdqtRcMQBfplZJgLCXrN+QuYaQLpz0k=";
+  };
+
+  propagatedBuildInputs = [
+    gitpython
+    mkdocs
+    pytz
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix:28]: Please add `pytestCheckHook` to `nativeCheckInputs` or `doCheck = false` with a comment.

File: pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  pythonImportsCheck = [ "mkdocs_git_revision_date_localized_plugin" ];
+
+  meta = with lib; {
```
Comment (natsukium) [inline pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix:31]: How about adding `meta.changelog`?
It's helpful for future updates.

File: pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix (hunk 2 above)
Comment (totoroot) [reply pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix:31]: Good idea, I added tests.

File: pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix (hunk 3 above)
Comment (totoroot) [reply pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix:40]: Good idea, done!

**Comment by natsukium** [comment]
Would you please change `pythonPackages` to `python310Packages` in the commit message and PR title?
pythonPackages means python2.

File: pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix (hunk 1 above)
Comment (totoroot) [reply pkgs/development/python-modules/mkdocs-git-revision-date-localized-plugin/default.nix:14]: Done!

**Comment by totoroot** [comment]
> Would you please change `pythonPackages` to `python310Packages` in the commit message and PR title? pythonPackages means python2.

I changed it to `python3Packages` as the package not only builds for Python 3.10.

**Comment by natsukium** [comment]
@ofborg build python310Packages.mkdocs-git-revision-date-localized-plugin

**Review by Janik-Haag — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

I'm not sure, ofborg doesn't build it with python3Packages for some reason.🤔

**Review by Janik-Haag — COMMENTED** [review]
`nix-build -A python3Packages.mkdocs-git-revision-date-localized-plugin` also doesn't work :thinking:

**Comment by totoroot** [comment]
@ofborg build python310Packages.mkdocs-git-revision-date-localized-plugin

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 242851` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.mkdocs-git-revision-date-localized-plugin</li>
    <li>python310Packages.mkdocs-git-revision-date-localized-plugin.dist</li>
    <li>python311Packages.mkdocs-git-revision-date-localized-plugin</li>
    <li>python311Packages.mkdocs-git-revision-date-localized-plugin.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
> nix-build -A python3Packages.mkdocs-git-revision-date-localized-plugin also doesn't work 🤔

It seems to work fine on both x86_64-linux and aarch64-darwin.
It may be a temporary error.

**Pull request merged**